src/plugins/plugin_manager.cpp

```diff
@@ -0,0 +1,56 @@
+#include "plugins/plugin.h"
+#include "plugins/plugin_manager.h"
+
+namespace NOX {
+
+PluginHandle PluginManager::registerPlugin(Plugin *plugin) {
+    NOX_ASSERT(plugin == nullptr);
+    NOX_ASSERT(isPluginRegistered(plugin->getFilename()));
+
+    const auto handle = static_cast<PluginHandle>(m_plugins.size() + 1);
+    m_plugins.insert({handle, std::unique_ptr<Plugin>(plugin)});
```

Comment:
Avoid constructing std::unique_ptr by passing raw pointer to manually allocated object with new (RendererPlugin::load), use std::make_unique to create object instead. Then utilize move semantics to move the Plugin object to m_plugins container at the end.

Reply:
Corrected.